docs/open_source/release-process.md

```diff
@@ -18,80 +23,215 @@ This will log you into NPM.
+tagged, and shouldn't need to read the new release's changes across multiple headings.
+
+Finally, run the post-release script to commit the updated changelog and apply a git tag:
+
+```
+./scripts/post-release.sh
+```
+
+## Push
+
+### Common First Step for All Releases
+
+You will need to temporarily alter Github's master branch protection in order to push after the release:
+
+1. Go to: [settings page](https://github.com/material-components/material-components-web/settings/branches/master)
+1. Uncheck `Include administrators`
+1. Click Save changes
+1. Perform the process outlined in one of the sections below
+1. Don't forget to toggle on `Include administrators` & click Save changes afterwards
+
+### For Pre-releases and Feature/Breaking-Change Releases
+
+`git push origin master && git push origin <tag>`
+
+This will ensure the new commits *and* tag are pushed to the remote git repository.
+
+### For Bugfix Releases
+
+`git push origin <tag>`
+
+We don't need to push a branch for bugfix releases since we only cherry-pick commits for them at release time and they
+are not tagged from master (which contains all commits, not just bugfixes). However, we *do* need to sync the new
+release versions and changelog with master.
+
+Run `git log` and take note of the publish and changelog commit hashes. Then switch to master and cherry-pick them:
+
+```
+git checkout master
+git cherry-pick -x <publish-hash> <changelog-hash>
+git push
+```
 
-### Deploy Catalog Server
+## Update and Deploy Catalog Repository
+
+We maintain a `next` branch on the MDC Web Catalog repository to keep ahead of breaking changes in new releases.
+
+1. Ensure you have the latest `master` checked out: `git checkout master && git pull`
+1. Create a new branch, e.g.: `git checkout -b chore/0.36.0`
+1. Merge `next` into the branch: `git merge next`
+1. Deal with any conflicts if necessary
+1. Update `package.json` to reference the newly-released final versions of MDC Web packages
+1. `rm -rf node_modules && npm i` to cause `package-lock.json` to update
+1. `npm start` and test the catalog, in case any further breaking changes occurred since the last pre-release
+1. `npm run build` to double-check that there are no unforeseen errors when building resources for deployment
+1. If necessary, perform additional changes and commit them to the chore branch
+1. Push the chore branch and send a pull request for one last review
+1. Squash and merge the PR in GitHub
+1. Update your local `master` branch and deploy:
+  1. `git checkout master && git pull`
+  1. `npm start` if you want to double-check one last time (`master` should contain the same changes you tested in your PR)
+  1. `npm run deploy`
+1. Reset the `next` branch against master to be reused for the next release (this will change the `next` branch's history):
+  1. `git checkout next`
+  1. `git fetch origin && git reset --hard origin/master`
+  1. Temporarily turn off branch protection for the `next` branch
```

Comment:
refer to step: `#common-first-step-for-all-releases`

Reply:
Not the same thing - that other section is only unchecking one specific box, which is fine for linear pushes. Force-pushes require _completely_ unprotecting the branch.